src/utils/logger.hpp

```diff
@@ -2,23 +2,58 @@
 
 #include <fmt/printf.h>
 
-extern int (*nrnpy_pr_stdoe_callback)(int, char*);
+using LoggerCallback = int(int, char*);
 
+class Logger {
+  public:
+    void setCallback(LoggerCallback* cb) {
+        callback = cb;
+    }
+
+    LoggerCallback* getCallback() const {
+        return callback;
+    }
+
+    template <typename... Args>
+    void info(const char* fmt, Args... args) {
+        if (callback) {
+            std::string message = fmt::format(fmt, std::forward<Args>(args)...);
+            callback(1, message.data());
+        }
+        fmt::print(fmt, std::forward<Args>(args)...);
+    }
+
+    template <typename... Args>
+    void error(const char* fmt, Args... args) {
+        if (callback) {
+            std::string message = fmt::format(fmt, std::forward<Args>(args)...);
+            callback(2, message.data());
+        }
+        fmt::print(stderr, fmt, std::forward<Args>(args)...);
+    }
+
+  private:
+    LoggerCallback* callback;
+};
+
+extern Logger logger;
+
+// Deprecated but there is ~700 to change so let it go a bit more
 template <typename... Args>
 int Fprintf(FILE* stream, const char* fmt, Args... args) {
-    if (nrnpy_pr_stdoe_callback && (stream == stdout || stream == stderr)) {
+    if (logger.getCallback() && (stream == stdout || stream == stderr)) {
         std::string message = fmt::sprintf(fmt, std::forward<Args>(args)...);
-        nrnpy_pr_stdoe_callback(stream == stdout ? 1 : 2, message.data());
+        logger.getCallback()(stream == stdout ? 1 : 2, message.data());
         return message.size();
     }
     return fmt::fprintf(stream, fmt, args...);
 }
```

Comment:
This doesn't seem finished yet. One improvement would be to allow the Logger to `print` to `FILE*`s, e.g.:
```
class Logger {
    void print(FILE* stream, char const * fmt, Args... args) {
        auto message = fmt::format(fmt, args);
        callback(stream, message.c_str());
     }

     void error(const char* fmt, Args... args) {
         print(stderr, fmt, args...);
     }      
};
```

Reply:
In fact, this is not used in the code and should not arrive in a logger. I can add a really easy class for writing to file (but I repeat `Fprintf` is not used to day to do that ^^).